test/mocks/setup.go

```diff
@@ -49,6 +50,7 @@ var MEVMCoreClient *evm_client.MockEVMCoreClient
 var MHTTPClient *http_client.MockHttpClient
 var MReadOnlyService *service.MockReadOnlyService
 var MEVMClient *evm_client.MockEVMClient
+var MEVMTokenClient *evm_token_client.MockEVMTokenClient
```

Comment:
These here can all be interfaces

and later given concrete mock implementations in the setup below

Reply:
Please open an issue for refactoring all of those types.